### PR TITLE
Incorporate MCODE ES6 Classes

### PR DESCRIPTION
**Pull request by danlee1025**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

Address JIRA-1544.

Greg, Karen, Nicole, and I all worked on this task.

- Added new MCODE and updated SHR ES6 classes to project.
- Updated wrappers and factories.
- We kept some ES6 classes that were needed in Flux but are no longer in MCODE/SHR.
- Updated all patient JSONs(`HardcodedPatient`, `HardCodedPatientMidYearDemo18`, `HardCodedSarcomaPatient`, and `TestPatient`) to use MCODE and updated SHR entry types.
- Tested all of our endpoints.  Ran through pre-release test script for `demo1`, `demo2`, `pilot1`, and `ccp`.
- Fixed/updated all tests that were broken after model changes.

Please thoroughly test the app since these changes are massive and may affect all parts of the app.  All functionality should be the same as `master`.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [=x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [x] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [x] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [x] Document any new APIs/extension points
- [x] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
-  Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/model/shr/adverse/ShrAdverseObjectFactory.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { getNamespaceAndName } from '../../json-helper';
-import NoAdverseEvent from './NoAdverseEvent';
+// import NoAdverseEvent from './NoAdverseEvent';
```
Comment (Dtphelan1) [inline src/model/shr/adverse/ShrAdverseObjectFactory.js]: SWRT?

File: src/model/shr/adverse/ShrAdverseObjectFactory.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 import AdverseEvent from './AdverseEvent';
-import SeriousAdverseEvent from './SeriousAdverseEvent';
+// import SeriousAdverseEvent from './SeriousAdverseEvent';
```
Comment (Dtphelan1) [inline src/model/shr/adverse/ShrAdverseObjectFactory.js]: SWRT?

File: src/model/shr/adverse/ShrAdverseObjectFactory.js [hunk 3]
```diff
@@ -24,9 +24,9 @@ export default class ShrAdverseObjectFactory {
       throw new Error(`Unsupported type in ShrAdverseObjectFactory: ${type}`);
     }
     switch (elementName) {
-      case 'NoAdverseEvent': return NoAdverseEvent.fromJSON(json);
+      //   case 'NoAdverseEvent': return NoAdverseEvent.fromJSON(json);
```
Comment (Dtphelan1) [inline src/model/shr/adverse/ShrAdverseObjectFactory.js]: SWRT? and the one on line 29

File: src/patient/PatientRecord.jsx [hunk 4]
```diff
@@ -227,8 +226,9 @@ class PatientRecord {
     }
 
     getPerson() {
+//        console.log(this.patient.person);
```
Comment (Dtphelan1) [inline src/patient/PatientRecord.jsx]: SWRT?

File: src/patient/PatientRecord.jsx [hunk 5]
```diff
@@ -814,19 +813,19 @@ class PatientRecord {
...
         let panels = this.getEntriesOfType(FluxGastrointestinalStromalTumorCancerGeneticAnalysisPanel);
-        panels.sort(this._clinicallyRelevantTimeTimeSorter);
+        panels.sort(this._relevantTimeTimeSorter);
         return panels;
     }
 
     getPathologyReportsChronologicalOrder() {
         let reports = this.getPathologyReports();
-        //reports.sort(this._clinicallyRelevantTimeTimeSorter);
+        //reports.sort(this._relevantTimeTimeSorter);
```
Comment (Dtphelan1) [inline src/patient/PatientRecord.jsx]: Should we uncomment this? If not, do we know what the justification was for commenting out in the first place? I know you weren't the originator of this, but thought I'd ask.

File: src/summary/metadata/SarcomaSummarySection.jsx [hunk 6]
```diff
@@ -25,20 +24,20 @@ export default class SarcomaSummarySection extends MetadataSection {
...
-                    ]
-                },
-                {
-                    defaultTemplate: "Tumor is ${.Tumor Size}.",
-                    dataMissingTemplate: "Tumor is ${.Tumor Size}.",
-                    useDataMissingTemplateCriteria: [
-                        ".Tumor Size"
-                    ]
-                },
+                // {
```
Comment (Dtphelan1) [inline src/summary/metadata/SarcomaSummarySection.jsx]: SWRT 27- 40?

File: src/summary/metadata/SarcomaSummarySection.jsx [hunk 7]
```diff
@@ -172,32 +171,32 @@ export default class SarcomaSummarySection extends MetadataSection {
                             },
                             // shortcut: "@stage"
```
Comment (Dtphelan1) [inline src/summary/metadata/SarcomaSummarySection.jsx]: Should we remove?

File: src/summary/metadata/SarcomaSummarySection.jsx [hunk 8]
```diff
@@ -172,32 +171,32 @@ export default class SarcomaSummarySection extends MetadataSection {
...
-                                const list = currentConditionEntry.getObservationsOfTypeChronologicalOrder(FluxTumorDimensions);
-                                if (list.length === 0) return null;
-                                const size = list.pop(); // last is most recent
-                                return  {   value: size.quantity.value + " " + size.quantity.unit, 
-                                            isUnsigned: patient.isUnsigned(size), 
-                                            source: this.determineSource(patient, size)
-                                        };
-                    },
-                        },
+                        // {
```
Comment (Dtphelan1) [inline src/summary/metadata/SarcomaSummarySection.jsx]: SWRT 174-199? Also, looks like there was a justification for removing tumor and mitotic division rate, could you talk to that a bit?

Comment (danlee1025) [reply src/summary/metadata/SarcomaSummarySection.jsx]: The tumor and mitotic division rate got moved to recent lab results.  We didn't want duplicate information in the summary section so we removed it.

File: src/model/oncology/FluxBreastCancer.js [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
         this._patientRecord = patientRecord;
-        this._condition = BreastCancer.fromJSON(json);
+        this._entry = this._condition = this._cancerDisorder = CancerDisorder.fromJSON(json);
     }
 
     // Returns a list of all dates that may be considered a diagnosisDate for this condition
     // getPotentialDiagnosisDates() {
     //     // TODO: Aggregate the relevant candidate dates for diagnosis
-    //     // console.log(this._condition)
+    //     // console.log(this._cancerDisorder)
```
[the remark this answers is not in the document]
Comment (danlee1025) [reply src/model/oncology/FluxBreastCancer.js]: Removed, it looks like `getPotentialDiagnosisDates` is inherited from a Child Class.

File: src/model/oncology/FluxOncologyObjectFactory.js [hunk 10]
```diff
@@ -42,7 +42,7 @@ export default class FluxOncologyObjectFactory {
             case 'KITVariant': return new FluxKITVariant(json, patientRecord);
             case 'PDGFRAVariant': return new FluxPDGFRAVariant(json, patientRecord);
             case 'MitoticRate': return new FluxMitoticRate(json, patientRecord);
-            default: return ShrOncologyObjectFactory.createInstance(json, type);
+            default: console.log("unknown Flux wrapper in oncology package: " + elementName); //return ShrOncologyObjectFactory.createInstance(json, type);
```
[the remark this answers is not in the document]
Comment (danlee1025) [reply src/model/oncology/FluxOncologyObjectFactory.js]: Changed to `console.error`.  I don't think the lack of return value messes anything up though.

File: src/model/oncology/FluxPDGFRAVariant.js [hunk 11]
```diff
@@ -1,11 +1,12 @@
...
+import Observation from '../shr/base/Observation';
 
 class FluxPDGFRAVariant {
     constructor(json) {
-        this._pdgfraVariant = GeneticVariant.fromJSON(json);
+        this._pdgfraVariant = Observation.fromJSON(json);
     }
     
     get abbreviatedName() {
+//        return this._kitVariant.subject.value.coding[0].displayText.value || 'PDGFRA';
```
[the remark this answers is not in the document]
Comment (danlee1025) [reply src/model/oncology/FluxPDGFRAVariant.js]: Looks like there is a separate `get value` to get the value property from the observation so `abbreviatedName` just returns a string?

File: src/patient/PatientRecord.jsx (hunk 5 above)
Comment (danlee1025) [reply src/patient/PatientRecord.jsx]: I just removed this line.  We have a JIRA to support multiple pathology reports.

File: src/model/oncology/FluxBreastCancer.js [hunk 12]
```diff
@@ -1,22 +1,22 @@
...
 import FluxSolidTumorCancer from './FluxSolidTumorCancer';
 import FluxTumorDimensions from '../oncology/FluxTumorDimensions';
+import CancerDisorder from '../mcode/CancerDisorder';
 
 class FluxBreastCancer extends FluxSolidTumorCancer {
     constructor(json, patientRecord) {
         super();
         this._patientRecord = patientRecord;
-        this._condition = BreastCancer.fromJSON(json);
+        this._entry = this._condition = this._cancerDisorder = CancerDisorder.fromJSON(json);
```
[the remark this answers is not in the document]
Comment (danlee1025) [reply src/model/oncology/FluxBreastCancer.js]: We usually need this if it extends a class and we need to use the inherited functions.  But in this case `this._condition` should be enough.

File: src/patient/PatientRecord.jsx (hunk 4 above)
Comment (Dtphelan1) [reply src/patient/PatientRecord.jsx]: Resolved!

File: src/summary/metadata/SarcomaSummarySection.jsx (hunk 8 above)
Comment (Dtphelan1) [reply src/summary/metadata/SarcomaSummarySection.jsx]: Gotcha!

File: src/patient/PatientRecord.jsx (hunk 5 above)
Comment (Dtphelan1) [reply src/patient/PatientRecord.jsx]: Sounds good !

File: src/model/oncology/FluxBreastCancer.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/model/oncology/FluxBreastCancer.js]: Cool!

File: src/model/oncology/FluxBreastCancer.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/model/oncology/FluxBreastCancer.js]: cool that sounds good toe me

File: src/model/oncology/FluxOncologyObjectFactory.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/model/oncology/FluxOncologyObjectFactory.js]: Just checking. Thanks for the heads up.

**Comment by nicoleng12** [comment]
This looks good. I tested all the endpoints and tested in both IE and chrome. I ran through the pre release scripts. @daniellee1025 is dylan's question about the line in the file: src/model/oncology/FluxPDGFRAVariant.js resolved or are you guys still discussing?

**Comment by Dtphelan1** [comment]
Looks good to me! I'm all set with all comments I made :) Great work everyone!!!

**Pull request merged**